djstripe/models.py

```diff
@@ -357,35 +368,92 @@ def can_charge(self):
             self.date_purged is None
 
     def has_active_subscription(self):
```

Comment:
Porting this method will be a bit more complicated. With one subscription, this method determines whether or not the user can log in or perform specific tasks.

Here are a few ideas to determine validity with the new schema:
1. Require a plan_id to be passed in. If the subscription that corresponds to that plan is valid, return True.
2. Figure out some way to denote which plan/subscription denotes access in settings.

From the use cases I've heard, people would like to use multiple subscriptions to grant access to different parts of the site depending on which plans users are subscribed to. The developer can then write custom logic to satisfy his/her needs. We can then offer a single default (# 2) if that logic isn't necessary, and base our optional middleware on that.

I suggest first going through the source and reviewing everything that touches this method in order to get a better picture of how this is used.